python/GangaLHCb/Lib/GaudiRunApp/GaudiRunAppUtils.py

```diff
@@ -0,0 +1,78 @@
+import subprocess
+import time
+from Ganga.Core.exceptions import GangaException
+from Ganga.Runtime.GPIexport import exportToGPI
+from Ganga.Utility.logging import getLogger
+from Ganga.Core.exceptions import ApplicationConfigurationError
+from .PythonOptsCmakeParser import PythonOptsCmakeParser
+
+logger = getLogger()
+
+def readInputData(optsfiles, app):
+    '''Returns a LHCbDataSet object from a list of options files. The
+       optional argument extraopts will decide if the extraopts string inside
+       the application is considered or not.
+
+    Usage example:
+    # Get the data from an options file and assign it to the jobs inputdata field
+    j.inputdata = readInputData([\"~/cmtuser/DaVinci_v22r0p2/Tutorial/Analysis/options/Bs2JpsiPhi2008.py\"], j.application)
+
+    This is also called behind the scenes for j.readInputData([\"~/cmtuser/DaVinci_v22r0p2/Tutorial/Analysis/options/Bs2JpsiPhi2008.py\"])
+
+    '''
+    if not isinstance(optsfiles, list):
+        optsfiles = [optsfiles]
+
+    # use a dummy file to keep the parser happy
+    if len(optsfiles) == 0:
+        raise GangaException("Need a file to parse to call this method")
+
+    try:
+        parser = PythonOptsCmakeParser(optsfiles, app)
+    except Exception as err:
+        msg = 'Unable to parse the job options. Please check options files and extraopts.'
+        logger.error("PythonOptsCmakeParserError:\n%s" % str(err))
+        raise ApplicationConfigurationError(None, msg)
+
+    return parser.get_input_data()
+
+exportToGPI('GetGaudiRunInputData', readInputData, 'Functions')
```

Comment:
Does this need to be exported, given that it's not documented anywhere? Is it ever expected to be used by the user directly?